convert2rhel/actions/conversion/preserve_only_rhel_kernel.py

```diff
@@ -0,0 +1,221 @@
+
+        loggerinst.info("Installing RHEL kernel ...")
+        output, ret_code = pkgmanager.call_yum_cmd(command="install", args=["kernel"])
+        kernel_update_needed = False
+
+        if ret_code != 0:
+            self.set_result(
+                level="ERROR",
+                id="FAILED_TO_INSTALL_RHEL_KERNEL",
+                title="Failed to install RHEL kernel",
+                description="There was an error while attempting to install the RHEL kernel from yum.",
+                remediations="Please check that you can access the repositories that provide the RHEL kernel.",
+            )
+            return
+
+        # Check if kernel with same version is already installed.
+        # Example output from yum and dnf:
+        #  "Package kernel-4.18.0-193.el8.x86_64 is already installed."
+        already_installed = re.search(r" (.*?)(?: is)? already installed", output, re.MULTILINE)
+        if already_installed:
+            rhel_kernel_nevra = already_installed.group(1)
+            non_rhel_kernels = pkghandler.get_installed_pkgs_w_different_fingerprint(
+                system_info.fingerprints_rhel, "kernel"
+            )
+            for non_rhel_kernel in non_rhel_kernels:
+                # We're comparing to NEVRA since that's what yum/dnf prints out
+                if rhel_kernel_nevra == pkghandler.get_pkg_nevra(non_rhel_kernel):
+                    # If the installed kernel is from a third party (non-RHEL) and has the same NEVRA as the one available
+                    # from RHEL repos, it's necessary to install an older version RHEL kernel and the third party one will
+                    # be removed later in the conversion process. It's because yum/dnf is unable to reinstall a kernel.
+                    info_message = (
+                        "Conflict of kernels: One of the installed kernels"
+                        " has the same version as the latest RHEL kernel."
+                    )
+                    loggerinst.info("\n%s" % info_message)
+                    self.add_message(
+                        level="INFO",
+                        id="CONFLICT_OF_KERNELS",
+                        title="Conflict of installed kernel versions",
+                        description=info_message,
+                    )
+                    pkghandler.handle_no_newer_rhel_kernel_available()
+                    kernel_update_needed = True
+
+        if kernel_update_needed:
+            pkghandler.update_rhel_kernel()
+
+
+class VerifyRhelKernelInstalled(actions.Action):
+    id = "VERIFY_RHEL_KERNEL_INSTALLED"
+    dependencies = ("INSTALL_RHEL_KERNEL",)
+
+    def run(self):
+        """Verify that the RHEL kernel has been successfully installed and raise an ERROR if not"""
+        super(VerifyRhelKernelInstalled, self).run()
+
+        loggerinst.info("Verifying that RHEL kernel has been installed")
+        if not pkghandler.is_rhel_kernel_installed():
```

Comment:
This function is only being used in this class, can we move this to this module?

Reply:
Actually, since it is so small, we can just copy it ove rhere and update the if statement

Reply:
@r0x0d so I should just move the function from pkghandler into the preserve_only_rhel_kernel module?

Reply:
Yeah, please

Reply:
We could get rid of the function as well, since it's just two lines, we could incorporate that inside the function itself. Or we can just bring it here, your call.